op-node/rollup/output_root.go

```diff
@@ -8,7 +8,7 @@ import (
 )
 
 // ComputeL2OutputRoot computes the L2 output root
-func ComputeL2OutputRoot(l2OutputRootVersion eth.Bytes32, blockHash common.Hash, blockRoot common.Hash, storageRoot common.Hash) eth.Bytes32 {
+func ComputeL2OutputRoot(l2OutputRootVersion eth.Bytes32, blockRoot common.Hash, storageRoot common.Hash, blockHash common.Hash) eth.Bytes32 {
```

Comment:
Did you check every usage of this function? The type are all the same so this has a large risk of being misued. This also now matches the usage in `HashOutputRootProof`, but that'd cause a mismatch.

Reply:
The only usage my LSP could find was within `op-node/node/api.go` on L117, and I updated that :+1: The `HashOutputRootProof` function was passing them in the order that they are concatenated for hashing, but the params for `ComputeL2OutputRoot` were in the wrong order for some reason.

Reply:
"wrong order" is in the eye of the beholder. If you properly supplied the function it'd hash them in the correct order.

I looked at the specs & `HashOutputRootProof` was wrong - it assumed that `ComputeL2OutputRoot` would take parameters in order that they'd be hashed. It's definitely a tricky API & probably should be changed, but that was not inherently incorrect.

We also don't appear to use `HashOutputRootProof` - I'm not sure if it was every used.

I'm fine making this change because I'm reasonably confident we found all usages of `ComputeL2OutputRoot`, but we've made a change to the external API where if someone updated it wouldn't throw a compiler error & instead by subtly wrong.

Reply:
> "wrong order" is in the eye of the beholder. If you properly supplied the function it'd hash them in the correct order.

Great point- by "wrong order," I meant that the order was "wrong" if you assume that the correct order of the parameters is the order in which they'd be concatenated for hashing (which in my head is the most intuitive).

`HashOutputRootProof` was unused prior to this, but there is a TODO comment to use it in the withdrawal migration script: https://github.com/ethereum-optimism/optimism/blob/develop/op-chain-ops/cmd/withdrawals/main.go#L907-L914 (cc @tynes)

Reply:
we should fix that in there. Separately, I didn't realize that we had tooling to do withdrawals in op-chain-ops.